rootfs/etc/nginx/lua/balancer/ewma.lua

```diff
@@ -102,21 +120,52 @@ function _M.sync(self, backend)
   self.traffic_shaping_policy = backend.trafficShapingPolicy
   self.alternative_backends = backend.alternativeBackends
 
-  local changed = not util.deep_compare(self.peers, backend.endpoints)
-  if not changed then
-    return
+  -- delete now old endpoints
+  local indexed_new_endpoints = {}
+  for _, endpoint in pairs(backend.endpoints) do
+    local endpoint_string = endpoint.address .. ":" .. endpoint.port
+    indexed_new_endpoints[endpoint_string] = true
+  end
+  for _, endpoint in pairs(self.peers) do
+    local endpoint_string = endpoint.address .. ":" .. endpoint.port
+    if not indexed_new_endpoints[endpoint_string] then
+      ngx.shared.balancer_ewma:delete(endpoint_string)
+      ngx.shared.balancer_ewma_last_touched_at:delete(endpoint_string)
+    end
+  end
+
+  -- Calculate max ewma, and set the ewma of
+  -- newly added endpoints to this value. This makes sure
+  -- we start sending requests slowly to the new endpoints.
+  local slow_start_ewma = 0
+  for _, endpoint in pairs(self.peers) do
+    local endpoint_string = endpoint.address .. ":" .. endpoint.port
+    local ewma = ngx.shared.balancer_ewma:get(endpoint_string) or 0
+    if slow_start_ewma < ewma then
+      slow_start_ewma = ewma
+    end
+  end
```

Comment:
Not sure whether this is the best heuristic. Maybe 75th percentile value would be better?